File: Decorator/PHP/Soy.php
<?php
  class Soy extends CondimentDecorator {
    public $beverage;

    public function __construct($beverage) {
      $this->description = " Soy";
      $this->beverage = $beverage;
    }

    public function getDescription() {
      return $this->beverage->getDescription() . $this->description;
    }

    public function cost() {
      return $this->beverage->cost() + 8;
    }
  }

File: Decorator/Go/decorator/soy.go
package decorator

type Soy struct {
  *Condiment
}

func NewSoy(beverage beverage) *Soy {
  return &Soy{&Condiment{beverage: beverage, price: 1.2, description: "Soy"}}
}

func (self *Soy) Cost() float64 {
  return self.beverage.Cost() + self.price
}

func (self *Soy) GetDescription() string {
  return self.beverage.GetDescription() + " with " + self.description
}
